wagtail/contrib/wagtailmodeladmin/forms.py

```diff
@@ -0,0 +1,26 @@
+from django import forms
+from django.utils.translation import ugettext as _
+from wagtail.wagtailcore.models import Page
+
+
+class PageChoiceField(forms.ModelChoiceField):
+    def label_from_instance(self, obj):
+        bits = []
+        for ancestor in obj.get_ancestors(inclusive=True).exclude(depth=1):
+            bits.append(ancestor.title)
+        return ' > '.join(bits)
+
+
+class ParentChooserForm(forms.Form):
+    parent_page = PageChoiceField(
+        label=_('Put it under'),
```

Comment:
"Put it under" isn't really consistant with wagtails tone in general. Perhaps "Choose parent page" would be more consistent with for example "Add child page" etc.